lib: updatehub: Don't build after convertion from DT_FLASH_AREA to FLASH_AREA macros

**Describe the bug**
samples/net/updatehub stop build after convertion from DT_FLASH_AREA to FLASH_AREA macros

**To Reproduce**
west build -b frdm_k64f samples/net/updatehub

**Expected behavior**
sample app build normally

**Screenshots or console output**
```
[116/189] Building C object zephyr/lib/updatehub/CMakeFiles/lib__updatehub.dir/updatehub.c.obj
FAILED: zephyr/lib/updatehub/CMakeFiles/lib__updatehub.dir/updatehub.c.obj 
ccache /home/gfbudke/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBOARD_FLASH_SIZE=CONFIG_FLASH_SIZE -DBUILD_VERSION=v2.3.0-rc1-99-ga0ae53410eef -DCPU_MK64FN1M0VLL12 -DKERNEL -DNDEBUG -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I../include -Izephyr/include/generated -I../soc/arm/nxp_kinetis/k6x -I../lib/libc/newlib/include -I../lib/fnmatch/. -I../subsys/net/lib/coap/. -I../subsys/net/lib/sockets/. -I../subsys/net/lib/dns/. -I../subsys/net/l2 -I/home/gfbudke/zephyros/mainline/modules/hal/cmsis/CMSIS/Core/Include -I/home/gfbudke/zephyros/mainline/modules/hal/nxp/mcux/devices/MK64F12 -I/home/gfbudke/zephyros/mainline/modules/hal/nxp/mcux/components/phyksz8081 -I/home/gfbudke/zephyros/mainline/modules/hal/nxp/mcux/drivers/kinetis/. -I/home/gfbudke/zephyros/mainline/modules/crypto/tinycrypt/lib/include -Os -imacros/home/gfbudke/zephyros/mainline/zephyr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m4 -mthumb -mabi=aapcs -imacros/home/gfbudke/zephyros/mainline/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-address-of-packed-member -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/gfbudke/zephyros/mainline/zephyr/samples/net/updatehub=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/gfbudke/zephyros/mainline/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/gfbudke/zephyros/mainline=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=nano.specs -std=c99 -MD -MT zephyr/lib/updatehub/CMakeFiles/lib__updatehub.dir/updatehub.c.obj -MF zephyr/lib/updatehub/CMakeFiles/lib__updatehub.dir/updatehub.c.obj.d -o zephyr/lib/updatehub/CMakeFiles/lib__updatehub.dir/updatehub.c.obj   -c /home/gfbudke/zephyros/mainline/zephyr/lib/updatehub/updatehub.c
/home/gfbudke/zephyros/mainline/zephyr/lib/updatehub/updatehub.c: In function 'install_update':
/home/gfbudke/zephyros/mainline/zephyr/lib/updatehub/updatehub.c:443:26: warning: implicit declaration of function 'FLASH_AREA_ID' [-Wimplicit-function-declaration]
  443 |  if (boot_erase_img_bank(FLASH_AREA_ID(image_1)) != 0) {
      |                          ^~~~~~~~~~~~~
/home/gfbudke/zephyros/mainline/zephyr/lib/updatehub/updatehub.c:443:40: error: 'image_1' undeclared (first use in this function)
  443 |  if (boot_erase_img_bank(FLASH_AREA_ID(image_1)) != 0) {
      |                                        ^~~~~~~
/home/gfbudke/zephyros/mainline/zephyr/lib/updatehub/updatehub.c:443:40: note: each undeclared identifier is reported only once for each function it appears in
[119/189] Building C object zephyr/lib/updatehub/CMakeFiles/lib__updatehub.dir/updatehub_firmware.c.obj
FAILED: zephyr/lib/updatehub/CMakeFiles/lib__updatehub.dir/updatehub_firmware.c.obj 
ccache /home/gfbudke/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBOARD_FLASH_SIZE=CONFIG_FLASH_SIZE -DBUILD_VERSION=v2.3.0-rc1-99-ga0ae53410eef -DCPU_MK64FN1M0VLL12 -DKERNEL -DNDEBUG -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I../include -Izephyr/include/generated -I../soc/arm/nxp_kinetis/k6x -I../lib/libc/newlib/include -I../lib/fnmatch/. -I../subsys/net/lib/coap/. -I../subsys/net/lib/sockets/. -I../subsys/net/lib/dns/. -I../subsys/net/l2 -I/home/gfbudke/zephyros/mainline/modules/hal/cmsis/CMSIS/Core/Include -I/home/gfbudke/zephyros/mainline/modules/hal/nxp/mcux/devices/MK64F12 -I/home/gfbudke/zephyros/mainline/modules/hal/nxp/mcux/components/phyksz8081 -I/home/gfbudke/zephyros/mainline/modules/hal/nxp/mcux/drivers/kinetis/. -I/home/gfbudke/zephyros/mainline/modules/crypto/tinycrypt/lib/include -Os -imacros/home/gfbudke/zephyros/mainline/zephyr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m4 -mthumb -mabi=aapcs -imacros/home/gfbudke/zephyros/mainline/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-address-of-packed-member -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/gfbudke/zephyros/mainline/zephyr/samples/net/updatehub=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/gfbudke/zephyros/mainline/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/gfbudke/zephyros/mainline=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=nano.specs -std=c99 -MD -MT zephyr/lib/updatehub/CMakeFiles/lib__updatehub.dir/updatehub_firmware.c.obj -MF zephyr/lib/updatehub/CMakeFiles/lib__updatehub.dir/updatehub_firmware.c.obj.d -o zephyr/lib/updatehub/CMakeFiles/lib__updatehub.dir/updatehub_firmware.c.obj   -c /home/gfbudke/zephyros/mainline/zephyr/lib/updatehub/updatehub_firmware.c
/home/gfbudke/zephyros/mainline/zephyr/lib/updatehub/updatehub_firmware.c: In function 'updatehub_get_firmware_version':
/home/gfbudke/zephyros/mainline/zephyr/lib/updatehub/updatehub_firmware.c:13:28: warning: implicit declaration of function 'FLASH_AREA_ID' [-Wimplicit-function-declaration]
   13 |  if (boot_read_bank_header(FLASH_AREA_ID(image_0), &header,
      |                            ^~~~~~~~~~~~~
/home/gfbudke/zephyros/mainline/zephyr/lib/updatehub/updatehub_firmware.c:13:42: error: 'image_0' undeclared (first use in this function)
   13 |  if (boot_read_bank_header(FLASH_AREA_ID(image_0), &header,
      |                                          ^~~~~~~
/home/gfbudke/zephyros/mainline/zephyr/lib/updatehub/updatehub_firmware.c:13:42: note: each undeclared identifier is reported only once for each function it appears in
[121/189] Building C object zephyr/boards/arm/frdm_k64f/CMakeFiles/boards__arm__frdm_k64f.dir/pinmux.c.obj
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK 0.11.2
 - Commit SHA a0ae534

CC @otavio

--- a/lib/updatehub/updatehub.c
+++ b/lib/updatehub/updatehub.c
@@ -21,6 +21,7 @@ LOG_MODULE_REGISTER(updatehub);
 #include <power/reboot.h>
 #include <tinycrypt/sha256.h>
 #include <data/json.h>
+#include <storage/flash_map.h>
 
 #include "include/updatehub.h"
 #include "updatehub_priv.h"

--- a/lib/updatehub/updatehub_device.c
+++ b/lib/updatehub/updatehub_device.c
@@ -5,6 +5,8 @@
  */
 #include "updatehub_device.h"
 
+#include <string.h>
+
 bool updatehub_get_device_identity(char *id, int id_max_len)
 {
 	u8_t hwinfo_id[DEVICE_ID_BIN_MAX_SIZE];

--- a/lib/updatehub/updatehub_firmware.c
+++ b/lib/updatehub/updatehub_firmware.c
@@ -4,6 +4,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <storage/flash_map.h>
+
 #include "updatehub_firmware.h"
 
 bool updatehub_get_firmware_version(char *version, int version_len)

--- a/samples/net/updatehub/sample.yaml
+++ b/samples/net/updatehub/sample.yaml
@@ -1,0 +1,15 @@
+sample:
+  description: UpdateHub enterprise-grade Firmware Over-the-Air (FOTA)
+  name: UpdateHub
+tests:
+  sample.net.updatehub:
+    harness: net
+    tags: net wifi
+    depends_on: netif
+    build_only: true
+    platform_whitelist: frdm_k64f sam_v71_xult
+    extra_configs:
+      - CONFIG_UPDATEHUB_PRODUCT_UID="e4d37cfe6ec48a2d069cc0bbb8b078677e9a0d8df3a027c4d8ea131130c4265f"
+      - CONFIG_UPDATEHUB_POLL_INTERVAL=1
+      - CONFIG_UPDATEHUB_CE=y
+      - CONFIG_UPDATEHUB_SERVER="10.5.3.67"

